src/bp/core/services/telemetry/sdk-methods.ts

```diff
@@ -0,0 +1,188 @@
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-SDK'
+    this.interval = ms('1d')
+  }
+
+  protected async getStats() {
+    return {
+      ...buildSchema(await this.getServerStats(), 'server'),
+      event_type: 'custom_roles',
+      event_data: { schema: '1.0.0', SDKMethods: await this.SDKUsage() }
+    }
+  }
+
+  private async SDKUsage(): Promise<SDKUsageEvent> {
+    const bots = await this.botService.getBotsIds()
+    return { actions: await this.getActionUsages(bots), hooks: await this.getHooksUsages() }
+  }
+
+  private async getActionUsages(bots: string[]): Promise<SDKUsage> {
+    const rootFolder = 'actions'
+    const globalActionsNames = await this.ghostService
+      .global()
+      .directoryListing('/', `${rootFolder}/*.js`)
+      .map(path => path.split('/').pop() || '')
+
+    const global = await Promise.map(globalActionsNames, this.parseFile(`/${rootFolder}`))
+    const perBots = await Promise.reduce(bots, this.botActionsReducer(rootFolder), [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private botActionsReducer(
+    rootFolder: string
+  ): (parsedFilesAcc: ParsedFile[], botId: string) => Promise<ParsedFile[]> {
+    return async (parsedFilesAcc, botId) => {
+      const botActionsNames = await this.ghostService.forBot(botId).directoryListing(`/${rootFolder}`, '*.js')
+      const parsedFiles = await Promise.map(botActionsNames, this.parseFile(`/${rootFolder}`, { botId }))
+      return [...parsedFilesAcc, ...parsedFiles]
+    }
+  }
+
+  private async getHooksUsages(): Promise<SDKUsage> {
+    const hooksPayload = await getHooksLifecycle(this.botService, this.ghostService)
+    const global = await this.parseHooks(hooksPayload.global.filter(hook => hook.type === 'custom'))
+    const perBots = await Promise.reduce(hooksPayload.perBots, this.botHooksReducer(), [] as ParsedFile[])
+
+    return { global, perBots }
+  }
+
+  private botHooksReducer(): (parsedFilesAcc: ParsedFile[], botHooks: BotHooks) => Promise<ParsedFile[]> {
+    return async (parsedFilesAcc, botHooks: BotHooks) => {
+      const { botId, hooks } = botHooks
+      const parsedFiles = await this.parseHooks(
+        hooks.filter(hook => hook.type === 'custom'),
+        botId
+      )
+      return [...parsedFilesAcc, ...parsedFiles]
+    }
+  }
+
+  private async parseHooks(hooks: Hook[], botId?: string): Promise<ParsedFile[]> {
+    return Promise.map(hooks, async hook => {
+      const { lifecycle, enabled, path } = hook
+      const usageParams = { lifecycle, enabled }
+
+      if (botId) {
+        _.set(usageParams, 'botId', botId)
+      }
+
+      return this.parseFile('/hooks', usageParams)(path!)
+    })
+  }
+
+  private parseFile(rootFolder: string, usageParams?: UsageParams): (name: string) => Promise<ParsedFile> {
+    return async (name: string) => {
+      const file = await this.readFileAsString(rootFolder, name, _.get(usageParams, 'botId'))
```

Comment:
Avoid _.get, not necessary

Reply:
fixed: ea8ccef7154fbde172eb306b7fa7e0f733e5592f